portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,10 +32,18 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
  
-  private ArrayList<String> comments = new ArrayList<String>();
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
  
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.ASCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    
+    ArrayList<String> comments = new ArrayList<String>();
```

Comment:
This is more of an FYI, I would only make this change if it's easier for YOU to read.

Java 8 introduced streams, which can make these types of "build a collection" code blocks much more straightforward.

```
List<String> comments =
    results
        .asList()
        .stream() // get the object as a stream
        .filter(entity -> entity.hasProperty("message")) // adding this to show some additional functionality, this will only include entities that actually have that property set
        .map(entity-> (String)entity.getProperty("message")) // get the message from each entitiy
        .collect(Collectors.toList()); // collect the messages into a list
```

Also, as an example of my other nit, I honestly have no idea what type of list this is going to return, but I also don't care

Reply:
One other approach:
```
List<String> comments = new ArrayList<>();
results.asList().forEach(entity -> comments.add((String) entity.getProperty("message"));
```

Personally I generally perfer the streams approach, but in some cases this is more useful

Reply:
Also, I really wish entity had a "getMessage" method, as then the lambdas could be replaced with Entity::getMessage

ex:
```
.map(Entity::getMessage)
```